paramiko/proxy.py

```diff
@@ -96,9 +96,14 @@ def recv(self, size):
 
                 r, w, x = select([self.process.stdout], [], [], select_timeout)
                 if r and r[0] == self.process.stdout:
-                    buffer += os.read(
-                        self.process.stdout.fileno(), size - len(buffer)
-                    )
+                    b = os.read(
+                        self.process.stdout.fileno(), size - len(buffer))
+                    if not b:
+                        if buffer:
+                            return buffer
+                        raise ProxyCommandFailure(' '.join(self.cmd),
```

Comment:
When trying out this code, it looks like this exception is always getting raised (paramiko==2.4.1) after a command successfully exits.  I think this should be:
```
                if r and r[0] == self.process.stdout:
                    b = os.read(self.process.stdout.fileno(), size - len(buffer))
                    if not b:
                        return buffer
                    buffer += b
            return buffer
```